tools/st2ctl

```diff
@@ -93,19 +93,26 @@ then
 fi
 
 # Note: Scripts already call reload with "--register-<content>"
-# TODO: Update packs. actions to only pass in a resource name excluding # --register prefix
 if [ ${1} == "reload" -o ${1} == "clean" ]; then
+    ALLOWED_REGISTER_FLAGS=(--register-all --register-actions --register-aliases --register-policies --register-rules --register-sensors --verbose)
+    DEFAULT_REGISTER_FLAGS='--register-actions --register-aliases --register-sensors'
+
+    if [ ! -z ${2} ]; then
+        for arg in ${@:2}; do
+            if [[ " ${ALLOWED_REGISTER_FLAGS[*]} " != *" $arg "* ]]; then # argument not allowed
+                echo -e "\e[31mError: Invalid argument provided: ${arg} \e[0m\n"
+                print_usage
+                exit 1
+            fi
+        done
+    fi
+
     if [ -z ${2} ]; then
-        REGISTER_FLAGS="--register-sensors --register-actions --register-aliases"
```

Comment:
I'm just slightly worried, since I don't remember if we have any end to end tests for st2ctl.

If we don't, now it would be a good time to add some to st2tests repo.